shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Wallet.hs

```diff
@@ -101,3 +114,36 @@ getFilteredUTxO ss addrs =
   UTxO $ Map.filter (\(TxOut addr _) -> addr `Set.member` addrs) fullUTxO
   where
     UTxO fullUTxO = getUTxO ss
+
+-- | Get the (private) leader schedule for this epoch.
+--
+--   Given a private VRF key, returns the set of slots in which this node is
+--   eligible to lead.
+getLeaderSchedule ::
+  ( Crypto crypto,
+    VRF.Signable
+      (VRF crypto)
+      Seed
+  ) =>
+  Globals ->
+  ShelleyState crypto ->
+  ChainDepState crypto ->
+  KeyHash 'StakePool crypto ->
+  SignKeyVRF crypto ->
+  Set SlotNo
+getLeaderSchedule globals ss cds poolHash key = Set.filter isLeader epochSlots
+  where
+    isLeader slotNo =
+      let y = VRF.evalCertified () (mkSeed seedL slotNo epochNonce) key
+       in Map.notMember slotNo overlaySched
+            && checkLeaderValue (VRF.certifiedOutput y) stake f
+    stake = maybe 0 fst $ Map.lookup poolHash poolDistr
```

Comment:
we could instead case the whole function on this lookup, and return `empty` if the pool is missing. but I've no real preference.